### DISCUSSION TITLE
Snowflake release plan

### DISCUSSION THREAD

**Discussion by skripteria** [skripteria/wn-snowflake-plugin#14] [Announcements]
# Current release plan

So far this is what I have in mind:

### v1.0.2. done 

- improve content preview to include tyle color. image and file

### v1.0.3 done
- add support for mediafile and mediaimage
- types constraint 

### v.1.0.4 done
- extended parsing support

### v1.1.0

- add Seo support to allow site editors manage things such as meta keywords from Snowflake backend

### v1.2.0

- add menu support

**Comment by arvislacis** [comment]
About SEO support - what's your greater idea behind this feature?

In my opinion, meta information of pages already can be edited via Winter CMS page editor and if you need some extra fields, for example, for meta keywords then they probably can be just extended form fields of previously mentioned Winter CMS native meta tag editor...

Side note - if we are talking only about SEO keywords then I think they are losing their role in SEO ratings every year and for some search engines they are already optional (or at least they are detected automatically by search engine AI).

**Comment by arvislacis** [comment]
And also menu editor is available via Winter Pages plugin (https://github.com/wintercms/wn-pages-plugin)...

I am not sure if you need to reinvent the wheel for SEO and Menu functionality if it already exists pretty well.

**Reply by skripteria** [reply]
The idea behind the 2 features is that site editors can just see Snowflake and maybe one or more specialized plugins in the backend. So there will no unneeded complexity that may add confusion to them and they have control about everything they need.

if you think it through Snowflake does make the wn-pages-plugin obsolete, in my eyes this is a good thing since the Pages plugin did not deliver a good user interface, actually I think Pages could be one of the main reasons why Winter / October are not as widely used as they should. Reactivating Pages just for a tiny part of its functionality (Menu Support) is not very helpful in my eyes .

But this Snowflake scenario only works when site editors in fact do get access to everything that is considered to be important. The 2 parts I could identify so far is menu support and meta tag access - the Seo part.

Personally I absolutely agree: search engines are far beyond the level that they may be manipulated with the Meta Tags, but my personal experience tells me that there are customers religiously believing that they can. Maybe my personal experience doesnt reflect the real world in this point, Seo would be my lowest priority if it was just for me. 

I agree that both features dont have the Snowflake "touch" but are sort of boring standard functionality that has to be provided. If we find a good replacement without opening access to the Winter core CMS and the Pages plugin i would be happy

**Reply by arvislacis** [reply]
Overall I agree with you and think that these features would be useful and very nice thing to have but while developing SEO and Menu functionality some things should be considered and thought over:

- For Meta tags/fields there definitely would be some kind of another plugin component which one site developers place in the `<header>` tag so Snowflake can fill that with necessary information;
- Probably same goes also about Menu functionality (that there would be different component or syntax for this);
- In my opinion, Meta tag (SEO) and Menu functionality should be enabled/disabled via Snowflake plugin Settings - because maybe some site developers would use Snowflake together with other SEO and/or Winter Page plugins so Snowflake may cause conflicts etc. so it would be nice if site devs could choose to enable or disable these features if needed. Also talking about this - if there would be a lot of new options then for some site editors that also may be very complex - so enable/disable option would help also in this case. Or maybe there could be different access (view) permissions for SEO and Menu editor sections too.

**Reply by skripteria** [reply]
I think the frontend output could all be handled by the SF component, but we certainly will know more once devlopment has started.

I do like the idea to have both features optional, so it doesnt pollute the UI when not needed. The settings way of controlling it is my favourite since its the simpler way.

**Comment by skripteria** [comment]
I havent found any more parsing bugs, so v1.0.4 with extended parsing support is out now.

@arvislacis happy chrismas to you!

**Comment by arvislacis** [comment]
@skripteria Happy Christmas for you too and congrats on releasing new version of Snowflake plugin. :)
